Utils/openssutils.py

```diff
@@ -0,0 +1,56 @@
+import struct
+import base64
+
+
+def number_to_bytes(i):
+    """
+    Pack number into bytes. Retun as string.
+    """
+    result = []
+    while i:
+        result.append(chr(i & 0xFF))
+        i >>= 8
+    result.reverse()
+    return ''.join(result)
+
+
+def bits_to_string(a):
+    """
+    Return string representation of bits in a.
+    """
+    index = 7
+    s = ""
+    c = 0
+    for bit in a:
+        c = c | (bit << index)
+        index = index - 1
+        if index == -1:
+            # noinspection PyTypeChecker
+            s = s + struct.pack('>B', c)
+            c = 0
+            index = 7
+    return s
+
+
+def openssl_publickey_to_ssh(file):
+    """
+    Return base-64 encoded key appropriate for ssh.
+    """
+    from pyasn1.codec.der import decoder as der_decoder
```

Comment:
Python 2.7 on Ubuntu 18 by default doesn't ship with this module. What to do?

Reply:
This also means that when the user didn't provide an ssh-rsa public key, the existing behavior is to try to extract a public key from the certificate present under /var/lib/waagent and add it to authorized_keys under $HOME/.ssh

The old behavior was to catch the exception

ext_utils.add_extension_event(name=hutil.get_name(),
                                          op=constants.WALAEventOperation.Enable,
                                          is_success=False,
                                          message="(02100)Failed to reset ssh key.")

and carry on, but the extension status itself would show succeeded.

Reply:
Fixed old behavior in new [commit](https://github.com/Azure/azure-linux-extensions/pull/1097/commits/20aff780d34ba7e71c672692ddc18095366705dd)

but we still cannot convert non ssh-rsa public keys on python2.7